src/main/scala/se/nimsa/sbx/filtering/FilteringServiceActor.scala

```diff
@@ -0,0 +1,82 @@
+package se.nimsa.sbx.filtering
+
+import akka.actor.{Actor, Props}
+import akka.event.{Logging, LoggingReceive}
+import akka.util.Timeout
+import se.nimsa.sbx.app.GeneralProtocol.{Source, SourceDeleted, SourceRef}
+import se.nimsa.sbx.filtering.FilteringProtocol._
+import se.nimsa.sbx.util.FutureUtil.await
+
+import scala.concurrent.Future
+
+class FilteringServiceActor(filteringDAO: FilteringDAO)(implicit timeout: Timeout) extends Actor {
+  val log = Logging(context.system, this)
+
+  implicit val system = context.system
+  implicit val ec = context.dispatcher
+
+  val emptyTagPathsSeq: Seq[TagFilterTagPath] = Seq()
+
+  override def preStart(): Unit =
+    context.system.eventStream.subscribe(context.self, classOf[SourceDeleted])
+
+  def receive: Receive = LoggingReceive {
+    case AddTagFilter(tagFilterSpec) =>
+      sender ! insertTagFilter(tagFilterSpec)
```

Comment:
Seems like these cases are missing a response to sender. Is this work in progress? Even though the response is not used for some calls in the main code, it is great to always send a response for test purposes.